### PR TITLE
Update API

### PR DESCRIPTION
**Pull request by vgpechenkin**
- Figma API documentation is missing layoutGrow for FRAME node type
- Apply the latest Figma REST API changes (Includes AutoLayout, PluginData, Hyperlinks, ComponentSets aka Variants)
- Overall code consistency & typo fixes
- Deprecated meta

Thank you so much @neelts

I'm not sure if I should have run the build script

### CODE REVIEW

File: src/api-funcs.ts [hunk 1]
```diff
@@ -50,6 +50,8 @@ export function getFileApi(this: ApiClass,
         depth?: number,
         /** Set to "paths" to export vector data */
         geometry?: 'paths',
+        /** A comma separated list of plugin IDs and/or the string "shared". Any data present in the document written by those plugins will be included in the result in the `pluginData` and `sharedPluginData` properties. */
+        pluginData?: string,
```
Comment (didoo) [inline src/api-funcs.ts:54]: according to the documentation, is still `plugin_data`:
https://www.figma.com/developers/api#get-files-endpoint

have you tested it with the pascal-case parameter too?

(also at line `79` is `plugin_data`)

File: src/api-funcs.ts [hunk 2]
```diff
@@ -121,7 +129,7 @@ export function postCommentsApi(
     const body: any = {
         message,
         client_meta,
-        comment_id,
+        ...(comment_id && { comment_id })
```
Comment (didoo) [inline src/api-funcs.ts:132]: do you think there is a simpler way to make it optional? I imagine someone with a limited knowledge of JS, they could struggle to understand what is going on
alternatively, we could consider a small inline comment for this?

**Comment by didoo** [comment]
@vgpechenkin (and @neelts) first of all thank you so much for this PR, very thorough! 

I have added a couple of comments, let me know what you think.

One question: how did you do all this? checking manually all the definitions and comparing them with the documentation on https://www.figma.com/developers/api? or there is some automated way that I don't know of?

File: src/api-funcs.ts (hunk 1 above)
Comment (neelts) [reply src/api-funcs.ts:54]: you're right! we'll fix it

**Comment by neelts** [comment]
@didoo mostly manually, but parsing [developers.min.js](https://www.figma.com/figbuild-artifacts/developers.7ec24ed3409a5d96ca8583cdb14c50e9.min.js.br) also helps, tried this approach a few years ago, but I'm sure it's broken now [FigmaAPIExtract.hx](https://github.com/neelts/figma-api-hx/blob/master/src/figma/FigmaAPIExtract.hx)

File: src/api-funcs.ts (hunk 1 above)
Comment (vgpechenkin) [reply src/api-funcs.ts]: done

File: src/api-funcs.ts (hunk 2 above)
Comment (vgpechenkin) [reply src/api-funcs.ts]: I've replaced with a clearer expression

**Review by didoo — APPROVED** [review]

**Comment by didoo** [comment]
@vgpechenkin @neelts just published version `1.9.0` on nom. can you check if works for you? https://www.npmjs.com/package/figma-api/v/1.9.0 
thanks

**Comment by vgpechenkin** [comment]
@didoo I've replaced our fork with your version 1.9.0 and It works great. 
But I have one error with type prop spread in EffectShadow_.
` spread: Number;`
I think type Nmber needs to replace to number?
What do you think? 
thanks.

**Comment by didoo** [comment]
@vgpechenkin released version `1.9.1` to fix it: https://www.npmjs.com/package/figma-api/v/1.9.1

**Pull request merged**